src/transformers/models/t5/modeling_t5.py

```diff
@@ -1595,8 +1595,8 @@ def forward(
 
         # If decoding with past key value states, only the last tokens
         # should be given as an input
-        if past_key_values is not None:
-            assert labels is None, "Decoder should not use cached key value states when training."
+        if past_key_values is not None and use_cache:
+            # assert labels is None, "Decoder should not use cached key value states when training."
```

Comment:
I think this could be removed completely, it should be possible to pass `decoder_input_ids` of length more than 1 even when `past_key_values` are passed. During generation, the `prepare_inputs_for_generation`  method takes care of only feeding the last token, so this is not required here IMO. WDYT @patrickvonplaten

Reply:
Agree we can delete this!

Reply:
Should I make a new commit that only removes the whole if statement?  Or you could edit this commit directly?